src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -388,66 +399,75 @@ public UnfilteredPartitionIterator executeLocally(ReadExecutionController execut
     {
         long startTimeNanos = System.nanoTime();
 
-        ColumnFamilyStore cfs = Keyspace.openAndGetStore(metadata());
-        Index index = getIndex(cfs);
-
-        Index.Searcher searcher = null;
-        if (index != null)
+        COMMAND.set(this);
```

Comment:
the diff is large but it is just moving all the logic into the try block; no code change was done.